ambari-server/src/main/java/org/apache/ambari/server/orm/entities/HostStateEntity.java

```diff
@@ -137,7 +137,7 @@ public boolean equals(Object o) {
 
     HostStateEntity that = (HostStateEntity) o;
 
-    if (hostId != null ? hostId != that.hostId : that.hostId != null) return false;
+    if (hostId != null ? !hostId.equals(that.hostId) : that.hostId != null) return false;
```

Comment:
You might prefer using Objects.equals() here

Reply:
@virajjasani Thanks for your suggestion. It's done.